### DISCUSSION TITLE
Creating grid from cells

### DISCUSSION THREAD

**Discussion by kirillantonov2901** [pyvista/pyvista#1570] [Q&A]
Hi there! I'm newbie to pyvista but still trying to carry out my idea)
Goal is to plot (visualize) grid from geological simulator. Input data is vertices of cells + integer parameter - cell color (8 verticles for each cell) in XYZ parameter format. It contains several millions cells and I think I would be rather hard to plot the whole model so in my opinion it would be right to plot only it's  faces. But I have no idea to start from, need help. From examples it seems that unstructed grid is OK for my purpose. 
I'd like is to get this as a result
![Reservoir-model-used-for-history-matching-the-observed-changes-in-the-reservoir-showing](https://user-images.githubusercontent.com/55129933/128609112-79065d96-62df-4f70-a2fa-5a1adf087587.png)

**Comment by akaszynski** [comment]
Looks like this would be best represented as a [Rectilinear Grid](https://docs.pyvista.org/core/grids.html).

When visualizing within PyVista and VTK, the first step is to extract the external surface (which is done automatically).  This tends to scale really well; even huge models display quickly.

**Reply by banesullivan** [reply]
This should actually be a `StructuredGrid` due to the curvilinear structure. A `RectilinearGrid` cannot support this.

**Comment by banesullivan** [comment]
This should be very easily handled by PyVista, if you can share these data I would be happy to try to put an example together for you
